todo-mvc-kitchensink/src/widgets/TodoDetails.ts

```diff
@@ -24,29 +24,22 @@ class FocusableTextArea extends FocusableTextInput {
 @theme(styles)
 export default class TodoDetails extends I18nMixin(ThemeableMixin(WidgetBase))<TodoDetailsProperties> {
 	onClose() {
-		const { activeFilter: filter, activeView: view } = this.properties;
-
-		const closeLink = router.link(mainRoute, {
-			filter,
-			view
-		});
-
-		updateTodo(this.properties.todoDetails);
-		document.location.href = closeLink;
+		this.properties.updateTodo(this.properties.todo, this.properties.todo.id);
+		this.properties.showTodoDetails();
 	}
 
 	onInput(event: KeyboardEvent) {
-		this.properties.todoDetails.label = (<any> event.target).value;
+		this.properties.todo.label = (<any> event.target).value;
 	}
 
 	onCompleted() {
-		this.properties.todoDetails.completed = !this.properties.todoDetails.completed;
+		this.properties.todo.completed = !this.properties.todo.completed;
```

Comment:
doesn't this goes against number 5 of https://github.com/dojo/widget-core#key-principles?. If this wasn't nested in another object I think you would have got a `readonly` error.